server/src/main/java/org/elasticsearch/transport/TransportActionStats.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.transport;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public record TransportActionStats(
+    String action,
+    long requestCount,
+    long totalRequestSize,
+    long[] requestSizeHistogram,
+    long responseCount,
+    long totalResponseSize,
+    long[] responseSizeHistogram
+) implements Writeable, ToXContentFragment {
+
+    public TransportActionStats(StreamInput in) throws IOException {
+        this(in.readString(), in.readVLong(), in.readVLong(), in.readVLongArray(), in.readVLong(), in.readVLong(), in.readVLongArray());
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(action);
+
+        builder.startObject("requests");
+        builder.field("count", requestCount);
+        builder.humanReadableField("total_size_in_bytes", "total_size", ByteSizeValue.ofBytes(totalRequestSize));
+        histogramToXContent(builder, requestSizeHistogram, "histogram");
+        builder.endObject();
+
+        builder.startObject("responses");
+        builder.field("count", responseCount);
+        builder.humanReadableField("total_size_in_bytes", "total_size", ByteSizeValue.ofBytes(totalResponseSize));
+        histogramToXContent(builder, responseSizeHistogram, "histogram");
+        builder.endObject();
+
+        return builder.endObject();
+    }
+
+    static void histogramToXContent(XContentBuilder builder, long[] sizeHistogram, String fieldName) throws IOException {
```

Comment:
Sure, I think I used the shorter one to avoid some line breaking mess but it's not necessary any more. Done in e24b97e36d110f1674ed81c259b812b6ea738382.